py_trees/display.py

```diff
@@ -183,6 +212,33 @@ def generate_lines(root, internal_indent):
     return s
 
 
+def html_tree(
+        root,
+        show_status=False,
+        visited={},
+        previously_visited={},
+        indent=0,
+     ):
+    """
+    Graffiti your ascii tree on an html snippet.
+
+    Args:
+        root (:class:`~py_trees.behaviour.Behaviour`): the root of the tree, or subtree you want to show
+        show_status (:obj:`bool`): always show status and feedback message (i.e. for every element, not just those visited)
+        visited (dict): dictionary of (uuid.UUID) and status (:class:`~py_trees.common.Status`) pairs for behaviours visited on the current tick
+        previously_visited (dict): dictionary of behaviour id/status pairs from the previous tree tick
+        indent (:obj:`int`): the number of characters to indent the tree
+
+    Returns:
+        :obj:`str`: an ascii tree (i.e. as a html snippet)
+
+    .. seealso:: :meth:`py_trees.display.ascii_tree`
+    """
+    lines = ascii_tree(root, show_status, visited, previously_visited, indent, symbols=html_symbols)
```

Comment:
Given the simplicity, maybe just handle this in ascii_tree?

Reply:
I was coming at it from the user angle. They're going to find it easier to think of `ascii_tree` and `html_tree` without having to dig down into what symbols variable they should pass to it. It will also be more robust to breaking changes in the future - the symbol formatter is something that is quite likely to change, but the rest of it should be stable.

Having said that, I've only dragged myself halfway there. I really ought to have:

* private `_generate_text_tree` method that does the grunt work
* public `ascii_tree` method
* public `unicode_tree` method that falls back to `ascii_tree` when it discovers unicode is not available
* public `xhtml_tree` method